### PR TITLE
Cannot resolve .. [Errno -2] Name or service not known

### PR DESCRIPTION
**Pull request by ksarabu1**
Fixes #

## Why is this needed?

Zookeeper is running from Docker containers and the service being resolved using Consul. When there's a network issue, the call to `socket.getaddrinfo` in method `_expand_client_hosts` causing it to return the error `Cannot resolve .. [Errno -2] Name or service not known`. This is causing retry to throwback `STOP_CONNECTING` and it stops retrying as result.

## Proposed Changes

Change: Raise `ForceRetryError` if method `_expand_client_hosts` returns empty host/ports and `client.hosts` has only one host to resolve.

```
        # Check for an empty hostlist, indicating none resolved
        if len(host_ports) == 0:
            if len(self.client.hosts) == 1:
                raise ForceRetryError('Reconnecting')
            return STOP_CONNECTING
```

## Does this PR introduce any breaking change?
n/a

### CODE REVIEW

File: kazoo/protocol/connection.py [hunk 1]
```diff
@@ -543,6 +543,8 @@ def _connect_loop(self, retry):
 
         # Check for an empty hostlist, indicating none resolved
         if len(host_ports) == 0:
+            if len(self.client.hosts) == 1:
```
Comment (StephenSorriaux) [inline kazoo/protocol/connection.py:546]: Would you be willing to explain the reason for this additional condition? Looking at the code, I would have replaced the `return STOP_CONNECTING` with a `raise ForceRetryError('No host resolved, reconnecting')` but you may have seen something I am missing.

**Comment by StephenSorriaux** [comment]
Thans for thir PR.

Overall I am fine with the idea behind the change, it makes sense to trigger the retry policy in case hosts are not resolved.

**Comment by ceache** [comment]
I second @StephenSorriaux's feedback. The idea is good but I am confused by the additional check on a single host in the list.

File: kazoo/protocol/connection.py (hunk 1 above)
Comment (ksarabu1) [reply kazoo/protocol/connection.py:546]: Thanks. I was just trying to keep the behavior to default when multiple hosts are used and if they all fail to resolve. I would also prefer to keep it simple and raise retry in all cases. Will update.

**Review by StephenSorriaux — APPROVED** [review]
Thank you.

**Comment by ksarabu1** [comment]
@ceache  Can you please review.

**Review by ceache — APPROVED** [review]
Thanks!

**Comment by ksarabu1** [comment]
Thank you. Can you please merge the PR? Thanks in advance.

**Pull request merged**